cmd/argoexec/commands/wait.go

```diff
@@ -61,10 +59,24 @@ func waitContainer(ctx context.Context) error {
 	if err != nil {
 		wfExecutor.AddError(err)
 	}
-	// Annotating pod with output
-	err = wfExecutor.AnnotateOutputs(ctx, logArt)
-	if err != nil {
-		wfExecutor.AddError(err)
+
+	containerNames := wfExecutor.Template.GetMainContainerNames()
+	for _, containerName := range containerNames {
+		var art *v1alpha1.Artifact
+		if wfExecutor.Template.SaveLogsAsArtifact() {
+			// Saving logs
+			log.Info("Saving logs with new loop")
+			art, err = wfExecutor.SaveLogs(ctx, containerName)
+			if err != nil {
+				wfExecutor.AddError(err)
+			}
+		}
+
+		// Annotating pod with output
+		err = wfExecutor.AnnotateOutputs(ctx, art, containerName)
```

Comment:
this will result in N patches, but we only want to do this once, can you move out of the loop please?

Reply:
to get it outside of the loop i'd need to change `AnnotateOutputs` to accept an array of `Artifact/ContainerName` (maybe just add the `ContainerName` into the artifact struct?), update `AddAnnotation` to accept an array of `key/value` pairs, as well as `AddPodAnnotation` and `addPodMetadata`.
do you think there is a simpler way to do it?

maybe i should put all of the pod's outputs in a single annotation as a JSON object with the containerNames as keys? that might be simpler in that regard.